### PR TITLE
Convert features and their tests to Typescript

### PR DESCRIPTION
**Pull request by Wihito**
So far I've converted most of the features and their tests, along with some lib files.

### CODE REVIEW

File: src/js/types/deck.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export interface Card {
     type_line?: string;
     oracle_id?: string;
   };
+  color_identity: color[] | ["C"];
```
Comment (crookedneighbor) [inline src/js/types/deck.ts]: FYI, this is not correct.

This type is for the deck entry object on the Scryfall deckbuilder, but `color_identity` is a property that exists on the Scryfall API response, not the deck entry object.

File: test/features/card-input-modifier/index.test.ts [hunk 2]
```diff
@@ -1,20 +1,30 @@
-import bus from "framebus";
-import CardInputModifier from "Features/deck-builder-features/card-input-modifier";
-import deckParser from "Lib/deck-parser";
-import scryfall from "Lib/scryfall";
-import mutation from "Lib/mutation";
+import * as bus from "framebus";
+import CardInputModifier from "../../../src/js/features/deck-builder-features/card-input-modifier";
+import deckParser from "../../../src/js/lib/deck-parser";
```
Comment (crookedneighbor) [inline test/features/card-input-modifier/index.test.ts]: Any special reason these are not using the aliases? Maybe you were working on a branch before I added them to the tsconfig?

https://github.com/crookedneighbor/shambleshark/blob/master/tsconfig.json#L5-L16

File: src/js/types/deck.ts (hunk 1 above)
Comment (Wihito) [reply src/js/types/deck.ts]: Okay, I removed that line and the color type now

File: test/features/card-input-modifier/index.test.ts (hunk 2 above)
Comment (Wihito) [reply test/features/card-input-modifier/index.test.ts]: I think so. I'll look through the files and adjust them, and see if my IDE still does these wonky imports.

**Pull request merged**